ambry-frontend/src/main/java/com.github.ambry.frontend/AccountAndContainerInjector.java

```diff
@@ -72,7 +75,9 @@ void injectAccountAndContainerForPostRequest(RestRequest restRequest) throws Res
     } else if (frontendConfig.frontendAllowServiceIdBasedPostRequest) {
       ensureRequiredHeadersOrThrow(restRequest, requiredAmbryHeadersForPutWithServiceId);
       frontendMetrics.putWithServiceIdForAccountNameRate.mark();
-      injectAccountAndContainerUsingServiceId(restRequest);
+      String serviceId = getHeader(restRequest.getArgs(), Headers.SERVICE_ID, false);
```

Comment:
shouldn't required be `true` here?

Reply:
This was false before, but i think it should be true